contracts/core/VaultRouter.sol

```diff
@@ -421,12 +420,11 @@ contract VaultRouter is
     IERC4626 vault;
     uint256 shares;
     address receiver;
-    address owner;
+    address owner = msg.sender;
     assembly {
       vault := calldataload(data.offset)
       shares := calldataload(add(data.offset, 0x20))
       receiver := calldataload(add(data.offset, 0x40))
-      owner := calldataload(add(data.offset, 0x60))
     }
```

Comment:
Load owner from calldata and fallback to `msg.sender` if not provided. Use `msg.sender` as owner for backward compatibility.
```suggestion
    assembly {
        vault := calldataload(data.offset)
        shares := calldataload(add(data.offset, 0x20))
        receiver := calldataload(add(data.offset, 0x40))
        owner := calldataload(add(data.offset, 0x60))
    }

    owner = msg.sender;
```